translator/navjoy_translator.py

```diff
@@ -0,0 +1,347 @@
+                        version=f'{PROGRAM_NAME} {PROGRAM_VERSION}')
+    parser.add_argument('navjoyFile', help='navjoy file path')
+    parser.add_argument('--outputFile', required=False,
+                        default='navjoy_wzdx_translated_output_message.geojson', help='output file path')
+
+    args = parser.parse_args()
+    return args.navjoyFile, args.outputFile
+
+
+def wzdx_creator(messages, info=None, unsupported_message_callback=None):
+    if not messages:
+        return None
+   # verify info obj
+    if not info:
+        info = wzdx_translator.initialize_info(
+            '8d062f70-d53e-4029-b94e-b7fbcbde5885')
+    if not wzdx_translator.validate_info(info):
+        return None
+
+    wzd = wzdx_translator.initialize_wzdx_object(info)
+
+    for message in messages:
+        # closures can cover both directions
+        directions = message.get('data', {}).get('directionOfTraffic')
+
+        for direction in get_directions_from_string(directions):
+            # Parse closure to WZDx Feature
+            feature = parse_reduction_zone(
+                copy.deepcopy(message), direction, callback_function=unsupported_message_callback)
+            if feature:
+                wzd.get('features').append(feature)
+    if not wzd.get('features'):
+        return None
+    wzd = wzdx_translator.add_ids(wzd)
+    return wzd
+
+
+def get_linestring_index(map):
+    for i in range(len(map)):
+        if map[i].get("type") == "LineString":
+            return i
+
+
+def get_polygon_index(map):
+    for i in range(len(map)):
+        if map[i].get("type") == "Polygon":
+            return i
+
+
+def get_directions_from_string(directions_string) -> list:
+    if not directions_string or type(directions_string) != str:
+        return []
+
+    directions = []
+
+    # iterate over directions and convert short direction names to WZDx enum directions
+    directions_string = directions_string.strip()
+    for dir in directions_string.split('/'):
+        direction = STRING_DIRECTION_MAP.get(dir.lower())
+        if direction:
+            directions.append(direction)
+
+    return directions
+
+# TODO: Support additional zones per message (streetNameFrom2, ...)
+# Parse Navjoy  to WZDx
+
+
+def parse_reduction_zone(obj, direction, callback_function=None):
+    if not validate_closure(obj):
+        if callback_function:
+            callback_function(obj)
+        return None
+
+    data = obj.get('data', {})
+
+    map = data.get('srzmap')
+    index = get_linestring_index(map)
+    if index != None:
+        coordinates = map[index].get("coordinates", [None])[0]
+    else:
+        index = get_polygon_index(map)
+        if index != None:
+            polygon = map[index].get("coordinates", [None])[0]
+            coordinates = polygon_tools.polygon_to_polyline_center(polygon)
+        else:
+            logging.warning(
+                f"Invalid event with id = {obj.get('sys_gUid')}. No polygon or linestring")
```

Comment:
Should this really have just the "U" capitalized?

Reply:
yup

Reply:
Of course there's an oddity like that in the data haha